### ISSUE TITLE
Pipe Auto-run appears inconsistently applied.  What is the correct behavior?

### ISSUE THREAD

**Issue by wanderslth** [flexiodata/flexio-web-app#46]
Running pipes, I'm having trouble understanding what the rubric is for auto-run and preview along with the Save and Run Pipe buttons.

_NOTE:  I am guessing I am doing an improper command for convert here in the example but, regardless of that error/bug in particular, the input does not auto run and the steps apparently do.  Maybe?_

EXAMPLE 1:

1. Create a new pipe with a web url - https://raw.githubusercontent.com/flexiodata/sample-data/master/contacts-2016-11.csv

![auto1](https://cloud.githubusercontent.com/assets/25911712/23582598/875bdc36-00f3-11e7-8323-10db546360d2.png)

2.  Double click on the pipe

![auto2](https://cloud.githubusercontent.com/assets/25911712/23582600/90dede66-00f3-11e7-8b27-5d6a3764f652.png)

3. Click on new task and add convert the Save.   (oh, also delete the, sigh, ghost step that was added so I can see the task I just ran):

![auto3](https://cloud.githubusercontent.com/assets/25911712/23582605/ad5b76bc-00f3-11e7-91b6-e5fd17e558f4.png)

Populates with JSON (so, apparently it ran?)

4.  Hit Run Pipe:

![auto4](https://cloud.githubusercontent.com/assets/25911712/23582606/b09298ce-00f3-11e7-9dd1-e2c25f36b729.png)

Now it outputs a table for me.

So, to recap:
* Pipe is created with an input
* No running at all
* Convert step does run, [also adds friggin ghost step!] and gives me some auto-run preview (though wrong)
* Run Pipe runs through the pipe and gives me the correct preview.

>> Whatever path we choose, whether to auto-run or to not auto-run, we need this to be crystal clear to the user -- should be a binary choice.  If we want this option available, then make it a toggle like you have with schedule on/off.  "Autorun: on/off" etc.  You can work around binary choices.  You cannot work around chef's surprise.
